### PR TITLE
Convert from wallet to bech32 address or pubkey

### PR DESCRIPTION
**Pull request by popenta**
Introduced two new possible `--out-format` types: `bech32-address` and `public-key`. They are used to convert a wallet to a `bech32 address` or a `public key`.

Solves https://github.com/multiversx/mx-sdk-py-cli/issues/327

### CODE REVIEW

File: multiversx_sdk_cli/cli_wallet.py [hunk 1]
```diff
@@ -56,7 +63,7 @@ def setup_parser(args: List[str], subparsers: Any) -> Any:
     sub.add_argument("--infile", help="path to the input file")
     sub.add_argument("--outfile", help="path to the output file")
     sub.add_argument("--in-format", required=True, choices=WALLET_FORMATS, help="the format of the input file")
-    sub.add_argument("--out-format", required=True, choices=WALLET_FORMATS, help="the format of the output file")
+    sub.add_argument("--out-format", required=True, choices=WALLET_FORMATS_AND_ADDRESSES, help="the format of the output file")
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_wallet.py:66]: Now I see that `wallet bech32` can be absorbed into this command in the future, as well. E.g. `in-format=public-key, --out-format=bech32-address`.

File: multiversx_sdk_cli/cli_wallet.py [hunk 2]
```diff
@@ -20,13 +20,20 @@
...
+WALLET_FORMAT_PUBLIC_KEY = "public-key"
+
 WALLET_FORMATS = [
     WALLET_FORMAT_RAW_MNEMONIC,
     WALLET_FORMAT_KEYSTORE_MNEMONIC,
     WALLET_FORMAT_KEYSTORE_SECRET_KEY,
     WALLET_FORMAT_PEM,
 ]
 
+WALLET_FORMATS_AND_ADDRESSES: List[str] = []
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_wallet.py:33]: Could have also been ` = [...WALLET_FORMATS, WALLET_FORMAT_BECH32_ADDRESS, WALLET_FORMAT_PUBLIC_KEY]` (optionally).

File: multiversx_sdk_cli/cli_wallet.py [hunk 3]
```diff
@@ -209,6 +216,24 @@ def _create_wallet_content(
...
+        return address.bech32()
+
+    if out_format == WALLET_FORMAT_PUBLIC_KEY:
+        if mnemonic:
+            secret_key = mnemonic.derive_key(address_index)
+        assert secret_key is not None
+
+        pubkey = secret_key.generate_public_key()
+        address = pubkey.to_address(address_hrp)
+        return address.hex()  # type: ignore
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_wallet.py:235]: We can return directly `pubkey.hex()`, without `address = `.

**Review by andreibancioiu — APPROVED** [review]

File: multiversx_sdk_cli/cli_wallet.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_wallet.py:33]: Done

File: multiversx_sdk_cli/cli_wallet.py (hunk 3 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_wallet.py:235]: Indeed, done!

**Review by andreibancioiu — DISMISSED** [review]

File: multiversx_sdk_cli/cli_wallet.py [hunk 4]
```diff
@@ -20,13 +20,18 @@
 WALLET_FORMAT_KEYSTORE_MNEMONIC = "keystore-mnemonic"
 WALLET_FORMAT_KEYSTORE_SECRET_KEY = "keystore-secret-key"
 WALLET_FORMAT_PEM = "pem"
+WALLET_FORMAT_BECH32_ADDRESS = "bech32-address"
+WALLET_FORMAT_PUBLIC_KEY = "public-key"
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/cli_wallet.py]: As discussed, renamed to `address-bech32` and `address-hex`.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**